zip.go

```diff
@@ -171,12 +171,14 @@ func (z *zipEntry) Sys() interface{} {
 	return z.zf.FileHeader
 }
 
-// Gid returns the group ID of the entry
+// Gid is not supported for zip files. The used library does not provide
+// this information. The function returns the group ID of the current process.
 func (z *zipEntry) Gid() int {
-	return os.Getgid()
+	return 0
 }
 
-// Uid returns the user ID of the entry
+// Uid is not supported for zip files. The used library does not provide
+// this information. The function returns the user ID of the current process.
 func (z *zipEntry) Uid() int {
-	return os.Getuid()
+	return 0
 }
```

Comment:
This returns zero, unlike the comment, and unlike the `rar` implementation. Is this intentional? It seems reasonable to maybe return `-1` in all cases, and just not call `chown` when the value is `-1`? (Since zero could be a real value.) Or maybe for archive entries that don't support owner preservation, just don't implement this/use a separate interface? These are just some ideas, I'm not sure what the right thing is to do here..

Reply:
I see the point that the comment is not accurate and a perfect fit. I will update that. 

If we return `-1`, I have a bad feeling with that. 

Assume we have a generic extraction service that should preserve file permission whenever possible. That means we need to be `root` (or as [you mentioned above](https://github.com/hashicorp/go-extract/pull/106#discussion_r1887351900), root-like). We could extract `tar` archives that contains these information. 

After extraction, the ownership is updated by our implementation. Whenever the service extracts an archive that does not contain such information (`rar`/`zip`/`7zip`), we just create the file with the extracting user (potentially `root`) and `chown` it to the user that has extracted the archive. Since we need `root` permissions to perform a `chown`, and the extracted file is anyway created with `uid:0`, bc/ the extracting user needs to be `root`. 

I am happy to role-back and return `os.Getuid()` and `os.Getgid()`, but not willing to return `-1`. A `-1` will encounter an error on a complete different code path, and that would be super tough to debug. Furthermore, I am happy to write a log entry.

Reply:
I addressed this behaviour in https://github.com/hashicorp/go-extract/pull/106/commits/e1ce17d6308871fc5e11a68b909bed14ea553537 and inverted the attributes preservation logic in https://github.com/hashicorp/go-extract/pull/106/commits/03a9e842b7a9d00b2e0aee0e9009fa25d0a47e65.

Reply:
Remark to that, i did multiple tests and the directory permissions are not always deterministically set on windows. So these test cases are skipped.

Reply:
OK, thanks for the context, I will take this into account when I review.